packages/server/lib/cloud/api/scrub_url.ts

```diff
@@ -0,0 +1,13 @@
+const SENSITIVE_KEYS = Object.freeze(['x-amz-credential', 'x-amz-signature', 'Signature', 'AWSAccessKeyId'])
+
+export const scrubUrl = (url: string): string => {
+  const parsedUrl = new URL(url)
+
+  for (const [key, value] of parsedUrl.searchParams) {
+    if (SENSITIVE_KEYS.includes(key)) {
+      parsedUrl.searchParams.set(key, 'X'.repeat(value.length))
```

Comment:
Do we want to match the value length? Not sure if just the length of the string could give something away?